### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by Will-Helliwell**
1 - user can logout
2 - a non-user cannot access the profile page by typing the url (redirected to login)

### CODE REVIEW

File: spec/features/web_helper.rb [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
+  fill_in "password", with: "wrong"
+  click_button "Enter"
+end
+
+def login
+  visit "/login"
+  User.create(username: "test_username", email: "test@test.com", password: "test")
+  fill_in "email", with: "test@test.com"
+  fill_in "password", with: "test"
+  click_button "Enter"
```
Comment (calavell) [inline spec/features/web_helper.rb:38]: nice use of helper methods to keep the spec files more succinct!

File: app.rb [hunk 2]
```diff
@@ -59,5 +57,8 @@ class AbodenB < Sinatra::Base
     @listings = Listing.all
     erb :listings
   end
-
+  post "/logout" do
+    session.clear
```
Comment (ollienorman) [inline app.rb]: I like how neat this solution is!

File: app.rb [hunk 3]
```diff
@@ -59,5 +57,8 @@ class AbodenB < Sinatra::Base
...
+    redirect "/"
```
Comment (ollienorman) [inline app.rb]: Think maybe makes more sense to redirect to the login page rather than the root?

Comment (Will-Helliwell) [reply app.rb]: Agreed! Have updated

**Pull request merged**